pylint/checkers/classes.py

```diff
@@ -931,11 +938,25 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
                     ):
                         break
             else:
-                function_repr = f"{function_def.name}({function_def.args.as_string()})"
+                name_stack = []
+                outer_level_names = ""
+                curr = parent_scope
+                # Generate proper names for nested functions
+                while curr != node:
+                    name_stack.append(curr.name)
+                    curr = curr.parent.scope()
+
+                if name_stack:
+                    outer_level_names = f"{'.'.join(reversed(name_stack))}."
```

Comment:
You don't need to guard this with an `IF`. Join with an empty list will just return an empty string.
```py
'.'.join(reversed([]))  # -> ""
```